src/nodes.coffee

```diff
@@ -1279,12 +1279,18 @@ exports.Class = class Class extends Base
       result
 
   compileClassDeclaration: (o) ->
-    @ctor ?= @makeDefaultConstructor() if @externalCtor
+    @ctor ?= @makeDefaultConstructor() if @externalCtor or @boundMethods.length
     @ctor?.noReturn = true
 
+    @proxyBoundMethods o if @boundMethods.length
+
     o.indent += TAB
 
     result = []
+    if @assignParentRef
```

Comment:
if we have bound methods and the parent class expression should be cached (as determined by `setParentRef()`), turn the class expression into a `(ref = parent.class().expression, class Child extends ref {...})`-style expression